include/ck/tensor_operation/gpu/thread/threadwise_tensor_slice_transfer_v3r1.hpp

```diff
@@ -244,126 +244,63 @@ struct ThreadwiseTensorSliceTransfer_v3r1
-
-                static_for<0, SrcScalarPerVector / elem_op_vec_len, 1>{}([&](auto idx) {
-                    // apply the src elementwise op and convert to DstData under the hood if needed
-                    src_element_op_(op_r_v.template AsType<dst_elem_op_vec_t>()(idx),
-                                    src_vector_container.template AsType<src_elem_op_vec_t>()[idx]);
+            using VectorSizeLookupTable    = Tuple<Sequence<>,
+                                                Sequence<I1>,
+                                                Sequence<I2>,
+                                                Sequence<I2, I1>,
+                                                Sequence<I4>,
+                                                Sequence<I4, I1>,
+                                                Sequence<I4, I2>,
+                                                Sequence<I4, I2, I1>,
+                                                Sequence<I8>,
+                                                Sequence<I8, I1>,
+                                                Sequence<I8, I2>,
+                                                Sequence<I8, I2, I1>,
+                                                Sequence<I8, I4>,
+                                                Sequence<I8, I4, I1>,
+                                                Sequence<I8, I4, I2>,
+                                                Sequence<I8, I4, I2, I1>,
+                                                Sequence<I16>>;
+            using VectorOffsetsLookupTable = Tuple<Sequence<>,
+                                                   Sequence<I0>,
+                                                   Sequence<I0>,
+                                                   Sequence<I0, I2>,
+                                                   Sequence<I0>,
+                                                   Sequence<I0, I4>,
+                                                   Sequence<I0, I4>,
+                                                   Sequence<I0, I4, I6>,
+                                                   Sequence<I0>,
+                                                   Sequence<I0, I8>,
+                                                   Sequence<I0, I8>,
+                                                   Sequence<I0, I8, I10>,
+                                                   Sequence<I0, I8>,
+                                                   Sequence<I0, I8, I12>,
+                                                   Sequence<I0, I8, I12>,
+                                                   Sequence<I0, I8, I12, I14>,
+                                                   Sequence<I0>>;
+
+            static_for<0, VectorSizeLookupTable{}.At(SrcScalarPerVector).Size(), 1>{}(
+                [&](auto v_idx) {
+                    constexpr auto VectorLoadSize =
+                        VectorSizeLookupTable{}.At(SrcScalarPerVector).At(v_idx);
```

Comment:
What about ?
```
tuple_element_t<SrcScalarPerVector, VectorSizeLookupTable>::At(v_idx);
```
In above helper function you don't have to create temporary object of `VectorSizeLookupTable`, you entirely work on just type. I wonder whether this may save some registers if compiler would have problems with optimizing this.